dut/drivers.wake

```diff
@@ -0,0 +1,59 @@
+
+tuple SkeletalDriverPlan =
+  global DuhFile:             Path
+  global CompanyName:         String
+  global DeviceName:          String
+  global LocalMetalDirectory: String
+  global DUTPlan:             RocketChipDUTPlan
+
+global def makeSkeletalDriverPlan = SkeletalDriverPlan _ _ _ _ _
+
+def getDevicePlan deviceName =
+  def plans = subscribe skeletalDriverPlans
+  def samePlan aPlan =
+    if aPlan.getSkeletalDriverPlanDeviceName ==~ deviceName
+    then Some aPlan
+    else None
+  mapPartial samePlan plans
+  | head
+
+def skeletalPlanToPlan skeletalDriverPlan =
+  def objModel =
+    makeRTL (skeletalDriverPlan.getSkeletalDriverPlanDUTPlan)
+    | getDUTObjectModel
+    | prettyJSON
```

Comment:
why not just put an `ObjectModel` field in the `SkeletalDriverPlan`? The `DUTPlan` has many fields you shouldn't care about, and you need to run the whole generator flow to get object model from it.

Reply:
If you have already generated the rtl, it's pretty fast. If you haven't generated the RTL you wouldn't have it and you would have to generate the RTL. I'm not sure how we would expect the Object model to come from besides makeRTL.

Reply:
but this is running a big job that really isn't this function's job to run. what if the `DUTPlan` passed in isn't actually a config that the user want to be built? or if someone needs to generate drivers for an existing object model, but they don't have access to the `DUTPlan` used to make that object model?

Reply:
Then where is the object model coming from?  If the code isn't tied to the dut, what is the point? Do we just ask them questions like duh?

Reply:
Just assume whoever is calling this function has an object model. If the caller does not have one then they will call `makeRTL` or whatever. The code shouldn't be tied to the dut. it should only be tied to the object model.

Reply:
I second @albertchen-sifive's suggestion about having this function/Plan take in a path to an object model JSON file as a parameter and not construct the object model itself.

Reply:
I think you will make the customer code more complicated and prone to error.

But, I will change it.